### PR TITLE
[HZ-1007] Add offload flag to map-store-config

### PR DESCRIPTION
**Pull request by ahmetmircik**
main PR which will use this config change: https://github.com/hazelcast/hazelcast/pull/21651
client protocol changes PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/424

**Modification:**
- Add `offload` parameter to `MapStoreConfig`, it is used to enable/disable map-store interaction in partition threads. 
   Default value is true.

### CODE REVIEW

**Review by mdumandag — COMMENTED** [review]
LGTM overall. I went over the checklist https://github.com/hazelcast/hazelcast/wiki/Checklist-for-Hazelcast-config-changes#yaml-configuration-schema-update and I believe we are missing the following things

- AbstractDynamicConfigGeneratorTest
- AbstractConfigBuilderTest
- hazelcast-full-example.[xml&yaml]
- tests described in https://github.com/hazelcast/hazelcast/wiki/Checklist-for-Hazelcast-config-changes#yaml-configuration-schema-update

can you please make sure that they(and other items on the checklist) are addressed?

Also, will you send a separate PR to use this new configuration element?

**Comment by ahmetmircik** [comment]
@mdumandag hard to follow that guide but i have updated PR based on your comment, hope not missing somewhere critical.

**Comment by mdumandag** [comment]
I think with the recent changes, we cover almost all the checklist items. The only remaining piece is the YAML schema validation tests mentioned in https://github.com/hazelcast/hazelcast/wiki/Checklist-for-Hazelcast-config-changes#yaml-configuration-schema-update

**Comment by ahmetmircik** [comment]
@mdumandag can you please have a look again?

**Review by mdumandag — APPROVED** [review]

**Comment by ahmetmircik** [comment]
run-lab-run

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/MapStoreConfigHolder.java [hunk 1]
```diff
@@ -181,6 +198,7 @@ public static MapStoreConfigHolder of(MapStoreConfig config, SerializationServic
         holder.setWriteBatchSize(config.getWriteBatchSize());
         holder.setWriteCoalescing(config.isWriteCoalescing());
         holder.setWriteDelaySeconds(config.getWriteDelaySeconds());
+        holder.setOffload(config.isOffload());
```
Comment (TomaszGaweda) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/MapStoreConfigHolder.java:201]: shouldn't holder. isOffloadExists be set here as well?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/MapStoreConfigHolder.java:201]: no, it is static utility to copy existing map-store-config, there is no state for `isOffloadExists`

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/MapStoreConfigHolder.java [hunk 2]
```diff
@@ -57,6 +61,8 @@ public MapStoreConfigHolder(boolean enabled, boolean writeCoalescing, int writeD
         this.factoryImplementation = factoryImplementation;
         this.properties = properties;
         this.initialLoadMode = initialLoadMode;
+        this.offload = offload;
+        this.isOffloadExists = isOffloadExists;
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/MapStoreConfigHolder.java:65]: maybe what you suggest is better, but i'm following auto generated client protocols naming standard, this is also inline with all naming in the past for similar purposes, i think it is fine based on this perspective.

**Review by TomaszGaweda — APPROVED** [review]

**Review by frant-hartm — CHANGES_REQUESTED** [review]

**Review by frant-hartm — APPROVED** [review]
